ccl/changefeedccl: TestShowChangefeedJobsRedacted failed

ccl/changefeedccl.TestShowChangefeedJobsRedacted [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20288446?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20288446?buildTab=artifacts#/) on master @ [a4ad94ae1379daa20b467c89d09de27ba8e2356d](https://github.com/cockroachdb/cockroach/commits/a4ad94ae1379daa20b467c89d09de27ba8e2356d):


```
=== RUN   TestShowChangefeedJobsRedacted
    test_log_scope.go:171: test logs captured to: /artifacts/tmp/_tmp/875edb2a1d3b708cdd086a008bbab2b7/logTestShowChangefeedJobsRedacted1840918762
    test_log_scope.go:82: use -show-logs to present logs inline
    helpers_test.go:1383: making server as secondary tenant
    changefeed_stmt.go:1584: requireNoFeedsFail: unexpected error: recv msg error: grpc: context canceled [code 1/Canceled]
    show_changefeed_jobs_test.go:232: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/875edb2a1d3b708cdd086a008bbab2b7/logTestShowChangefeedJobsRedacted1840918762
--- FAIL: TestShowChangefeedJobsRedacted (1.73s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestShowChangefeedJobsRedacted.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-53159

ccl/changefeedccl.TestShowChangefeedJobsRedacted [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20331120?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20331120?buildTab=artifacts#/) on master @ [65c856a23c004f4c9e1e553000fb63396f369faf](https://github.com/cockroachdb/cockroach/commits/65c856a23c004f4c9e1e553000fb63396f369faf):


```
=== RUN   TestShowChangefeedJobsRedacted
    test_log_scope.go:171: test logs captured to: /artifacts/tmp/_tmp/50a8284f7c7f32a812f6eaeb52caed36/logTestShowChangefeedJobsRedacted4158448873
    test_log_scope.go:82: use -show-logs to present logs inline
    helpers_test.go:1389: making server as secondary tenant
    changefeed_stmt.go:1584: requireNoFeedsFail: unexpected error: recv msg error: grpc: error reading from server: read tcp 127.0.0.1:56286->127.0.0.1:41775: use of closed network connection [code 14/Unavailable]
    show_changefeed_jobs_test.go:232: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/50a8284f7c7f32a812f6eaeb52caed36/logTestShowChangefeedJobsRedacted4158448873
--- FAIL: TestShowChangefeedJobsRedacted (1.69s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestShowChangefeedJobsRedacted.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestShowChangefeedJobsRedacted [failed](https://mesolite.cluster.engflow.com/invocations/default/afe5b61d-1ad3-4981-a15c-7589661136cc?testReportRun=5&testReportShard=2&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [fe3f3462c904d4735fef481fff4536925d9d7a77](https://github.com/cockroachdb/cockroach/commits/fe3f3462c904d4735fef481fff4536925d9d7a77):


```
=== RUN   TestShowChangefeedJobsRedacted
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestShowChangefeedJobsRedacted2403971787
    test_log_scope.go:82: use -show-logs to present logs inline
    helpers_test.go:1391: making server as secondary tenant
    changefeed_stmt.go:1670: requireNoFeedsFail: unexpected error: receiving from node 1: recv msg error: grpc: grpc: the client connection is closing [code 1/Canceled]
    changefeed_stmt.go:1670: requireNoFeedsFail: unexpected error: receiving from node 1: recv msg error: grpc: grpc: the client connection is closing [code 1/Canceled]
    show_changefeed_jobs_test.go:232: -- test log scope end --
test logs left over in: outputs.zip/logTestShowChangefeedJobsRedacted2403971787
--- FAIL: TestShowChangefeedJobsRedacted (1.48s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=5</code>
 - <code>shard=2</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestShowChangefeedJobsRedacted.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestShowChangefeedJobsRedacted [failed](https://mesolite.cluster.engflow.com/invocations/default/d0ceb6b2-e6a5-4146-8de1-4d9183aedf08?testReportRun=5&testReportShard=2&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [fe5ec41bce97958b4904bf898beddfe43d3f4e13](https://github.com/cockroachdb/cockroach/commits/fe5ec41bce97958b4904bf898beddfe43d3f4e13):


```
=== RUN   TestShowChangefeedJobsRedacted
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestShowChangefeedJobsRedacted2634332712
    test_log_scope.go:82: use -show-logs to present logs inline
    helpers_test.go:1391: making server as secondary tenant
    changefeed_stmt.go:1670: requireNoFeedsFail: unexpected error: receiving from node 1: recv msg error: grpc: grpc: the client connection is closing [code 1/Canceled]
    show_changefeed_jobs_test.go:232: -- test log scope end --
test logs left over in: outputs.zip/logTestShowChangefeedJobsRedacted2634332712
--- FAIL: TestShowChangefeedJobsRedacted (1.58s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=5</code>
 - <code>shard=2</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestShowChangefeedJobsRedacted.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

Something interesting is that all of the subtests passed but not the main test
```
--- FAIL: TestShowChangefeedJobsRedacted (1.20s)
    --- PASS: TestShowChangefeedJobsRedacted/api_secret (0.09s)
    --- PASS: TestShowChangefeedJobsRedacted/sasl_password (0.05s)
    --- PASS: TestShowChangefeedJobsRedacted/ca_cert (0.06s)
    --- PASS: TestShowChangefeedJobsRedacted/shared_access_key (0.05s)
    --- PASS: TestShowChangefeedJobsRedacted/jobs (0.05s)
```

I don't think that this is a release blocker since it is a matter of not giving the changefeed enough time to shut down before the test is over. I added a manual changefeed cancel via `CANCEL ALL CHANGEFEEDS` and waiting on the status on that changefeed to become `canceled` with a 2 minute timeout, and it no longer has the error from the latest failed run. However I am seeing this following instead, which might not be a changefeed issue.
```
panic: found 1 short-living non-stopped monitors in flow-7108c09f
	flow-7108c09f 0 B
	    changefntr-mem 0 B
```

--- a/pkg/ccl/changefeedccl/show_changefeed_jobs_test.go
+++ b/pkg/ccl/changefeedccl/show_changefeed_jobs_test.go
@@ -24,7 +24,6 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/kv/kvserver"
 	"github.com/cockroachdb/cockroach/pkg/sql/catalog/catpb"
 	"github.com/cockroachdb/cockroach/pkg/sql/catalog/descpb"
-	"github.com/cockroachdb/cockroach/pkg/sql/execinfra"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
@@ -166,69 +165,63 @@ func TestShowChangefeedJobsRedacted(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	s, stopServer := makeServer(t)
-	defer stopServer()
+	testFn := func(t *testing.T, s TestServer, f cdctest.TestFeedFactory) {
+		sqlDB := sqlutils.MakeSQLRunner(s.DB)
+		sqlDB.Exec(t, `CREATE TABLE foo (a INT PRIMARY KEY, b STRING)`)
 
-	knobs := s.TestingKnobs.
-		DistSQL.(*execinfra.TestingKnobs).
-		Changefeed.(*TestingKnobs)
-	knobs.WrapSink = func(s Sink, _ jobspb.JobID) Sink {
-		if _, ok := s.(*externalConnectionKafkaSink); ok {
-			return s
+		const apiSecret = "bar"
+		const certSecret = "Zm9v"
+		for _, tc := range []struct {
+			name                string
+			uri                 string
+			expectedSinkURI     string
+			expectedDescription string
+		}{
+			{
+				name: "api_secret",
+				uri:  fmt.Sprintf("confluent-cloud://nope?api_key=fee&api_secret=%s", apiSecret),
+			},
+			{
+				name: "sasl_password",
+				uri:  fmt.Sprintf("kafka://nope/?sasl_enabled=true&sasl_handshake=false&sasl_password=%s&sasl_user=aa", apiSecret),
+			},
+			{
+				name: "ca_cert",
+				uri:  fmt.Sprintf("kafka://nope?ca_cert=%s&tls_enabled=true", certSecret),
+			},
+			{
+				name: "shared_access_key",
+				uri:  fmt.Sprintf("azure-event-hub://nope?shared_access_key=%s&shared_access_key_name=plain", apiSecret),
+			},
+		} {
+			t.Run(tc.name, func(t *testing.T) {
+				foo := feed(t, f, fmt.Sprintf(`CREATE CHANGEFEED FOR TABLE foo INTO '%s'`, tc.uri),
+					optOutOfMetamorphicEnrichedEnvelope{reason: "compares text of changefeed statement"})
+				defer closeFeed(t, foo)
+
+				efoo, ok := foo.(cdctest.EnterpriseTestFeed)
+				require.True(t, ok)
+				jobID := efoo.JobID()
+
+				var sinkURI, description string
+				sqlDB.QueryRow(t, "SELECT sink_uri, description from [SHOW CHANGEFEED JOB $1]", jobID).Scan(&sinkURI, &description)
+				replacer := strings.NewReplacer(apiSecret, "redacted", certSecret, "redacted")
+				expectedSinkURI := replacer.Replace(tc.uri)
+				expectedDescription := replacer.Replace(fmt.Sprintf(`CREATE CHANGEFEED FOR TABLE foo INTO '%s'`, tc.uri))
+				require.Equal(t, expectedSinkURI, sinkURI)
+				require.Equal(t, expectedDescription, description)
+			})
 		}
-		return &externalConnectionKafkaSink{sink: s, ignoreDialError: true}
-	}
-
-	sqlDB := sqlutils.MakeSQLRunner(s.DB)
-	sqlDB.Exec(t, `CREATE TABLE foo (a INT PRIMARY KEY, b STRING)`)
-
-	const apiSecret = "bar"
-	const certSecret = "Zm9v"
-	for _, tc := range []struct {
-		name                string
-		uri                 string
-		expectedSinkURI     string
-		expectedDescription string
-	}{
-		{
-			name: "api_secret",
-			uri:  fmt.Sprintf("confluent-cloud://nope?api_key=fee&api_secret=%s", apiSecret),
-		},
-		{
-			name: "sasl_password",
-			uri:  fmt.Sprintf("kafka://nope/?sasl_enabled=true&sasl_handshake=false&sasl_password=%s&sasl_user=aa", apiSecret),
-		},
-		{
-			name: "ca_cert",
-			uri:  fmt.Sprintf("kafka://nope?ca_cert=%s&tls_enabled=true", certSecret),
-		},
-		{
-			name: "shared_access_key",
-			uri:  fmt.Sprintf("azure-event-hub://nope?shared_access_key=%s&shared_access_key_name=plain", apiSecret),
-		},
-	} {
-		t.Run(tc.name, func(t *testing.T) {
-			createStmt := fmt.Sprintf(`CREATE CHANGEFEED FOR TABLE foo INTO '%s'`, tc.uri)
-			var jobID jobspb.JobID
-			sqlDB.QueryRow(t, createStmt).Scan(&jobID)
-			var sinkURI, description string
-			sqlDB.QueryRow(t, "SELECT sink_uri, description from [SHOW CHANGEFEED JOB $1]", jobID).Scan(&sinkURI, &description)
-			replacer := strings.NewReplacer(apiSecret, "redacted", certSecret, "redacted")
-			expectedSinkURI := replacer.Replace(tc.uri)
-			expectedDescription := replacer.Replace(createStmt)
-			require.Equal(t, expectedSinkURI, sinkURI)
-			require.Equal(t, expectedDescription, description)
+		t.Run("jobs", func(t *testing.T) {
+			queryStr := sqlDB.QueryStr(t, "SELECT description from [SHOW JOBS]")
+			require.NotContains(t, queryStr, apiSecret)
+			require.NotContains(t, queryStr, certSecret)
+			queryStr = sqlDB.QueryStr(t, "SELECT sink_uri, description from [SHOW CHANGEFEED JOBS]")
+			require.NotContains(t, queryStr, apiSecret)
+			require.NotContains(t, queryStr, certSecret)
 		})
 	}
-
-	t.Run("jobs", func(t *testing.T) {
-		queryStr := sqlDB.QueryStr(t, "SELECT description from [SHOW JOBS]")
-		require.NotContains(t, queryStr, apiSecret)
-		require.NotContains(t, queryStr, certSecret)
-		queryStr = sqlDB.QueryStr(t, "SELECT sink_uri, description from [SHOW CHANGEFEED JOBS]")
-		require.NotContains(t, queryStr, apiSecret)
-		require.NotContains(t, queryStr, certSecret)
-	})
+	cdcTest(t, testFn, feedTestForceSink("kafka"), feedTestNoExternalConnection)
 }
 
 func TestShowChangefeedJobs(t *testing.T) {

